67_RayQueryGeometry/main.cpp

```diff
@@ -0,0 +1,1088 @@
+				imageBarriers[0].subresourceRange = {
+					.aspectMask = IImage::EAF_COLOR_BIT,
+					.baseMipLevel = 0u,
+					.levelCount = 1u,
+					.baseArrayLayer = 0u,
+					.layerCount = 1u
+				};
+				imageBarriers[0].oldLayout = IImage::LAYOUT::UNDEFINED;
+				imageBarriers[0].newLayout = IImage::LAYOUT::TRANSFER_SRC_OPTIMAL;
+
+				imageBarriers[1].barrier = {
+				   .dep = {
+					   .srcStageMask = PIPELINE_STAGE_FLAGS::NONE,
+					   .srcAccessMask = ACCESS_FLAGS::NONE,
+					   .dstStageMask = PIPELINE_STAGE_FLAGS::BLIT_BIT,
+					   .dstAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT
+					}
+				};
+				imageBarriers[1].image = m_surface->getSwapchainResources()->getImage(m_currentImageAcquire.imageIndex);
+				imageBarriers[1].subresourceRange = {
+					.aspectMask = IImage::EAF_COLOR_BIT,
+					.baseMipLevel = 0u,
+					.levelCount = 1u,
+					.baseArrayLayer = 0u,
+					.layerCount = 1u
+				};
+				imageBarriers[1].oldLayout = IImage::LAYOUT::UNDEFINED;
+				imageBarriers[1].newLayout = IImage::LAYOUT::TRANSFER_DST_OPTIMAL;
+
+				cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, { .imgBarriers = imageBarriers });
+			}
+
+			{
+				IGPUCommandBuffer::SImageBlit regions[] = {{
+					.srcMinCoord = {0,0,0},
+					.srcMaxCoord = {WIN_W,WIN_H,1},
+					.dstMinCoord = {0,0,0},
+					.dstMaxCoord = {WIN_W,WIN_H,1},
+					.layerCount = 1,
+					.srcBaseLayer = 0,
+					.dstBaseLayer = 0,
+					.srcMipLevel = 0,
+					.dstMipLevel = 0,
+					.aspectMask = IGPUImage::E_ASPECT_FLAGS::EAF_COLOR_BIT
+				}};
+				
+				auto srcImg = outHDRImageViews[m_currentImageAcquire.imageIndex]->getCreationParameters().image.get();
+				auto scRes = static_cast<CDefaultSwapchainFramebuffers*>(m_surface->getSwapchainResources());
+				auto dstImg = scRes->getImage(m_currentImageAcquire.imageIndex);
+
+				cmdbuf->blitImage(srcImg, IImage::LAYOUT::TRANSFER_SRC_OPTIMAL, dstImg, IImage::LAYOUT::TRANSFER_DST_OPTIMAL, regions, ISampler::ETF_NEAREST);
+			}
+
+			// TODO: transition to present
+			{
+				IGPUCommandBuffer::SPipelineBarrierDependencyInfo::image_barrier_t imageBarriers[1];
+				imageBarriers[0].barrier = {
+				   .dep = {
+					   .srcStageMask = PIPELINE_STAGE_FLAGS::BLIT_BIT,
+					   .srcAccessMask = ACCESS_FLAGS::TRANSFER_WRITE_BIT,
+					   .dstStageMask = PIPELINE_STAGE_FLAGS::NONE,
+					   .dstAccessMask = ACCESS_FLAGS::NONE
+					}
+				};
+				imageBarriers[0].image = m_surface->getSwapchainResources()->getImage(m_currentImageAcquire.imageIndex);
+				imageBarriers[0].subresourceRange = {
+					.aspectMask = IImage::EAF_COLOR_BIT,
+					.baseMipLevel = 0u,
+					.levelCount = 1u,
+					.baseArrayLayer = 0u,
+					.layerCount = 1u
+				};
+				imageBarriers[0].oldLayout = IImage::LAYOUT::TRANSFER_DST_OPTIMAL;
+				imageBarriers[0].newLayout = IImage::LAYOUT::PRESENT_SRC;
+
+				cmdbuf->pipelineBarrier(E_DEPENDENCY_FLAGS::EDF_NONE, { .imgBarriers = imageBarriers });
+			}
```

Comment:
btw example 27 stores straight to swapchain IIRC

Reply:
still TODO

Reply:
Looks like a blit is used as well in example 27

Reply:
ah yes, indeed probably because the example is meant to do simple Whitted Raytracing with accumulation and most desktops can't be relied upon to have a HDR swapchain